content/docs/details/scaffolder/index.md

```diff
@@ -0,0 +1,204 @@
+
+A Scaffolder _template_ is a configurable process that will run one or more Scaffolder _steps_. The template will be run when a user visits the "Create Component" page in Backstage. `https://<tenant-name>.roadie.so/create`.
+
+![create-a-new-component](./create-a-new-component.png)
+
+Templates are defined by a Backstage Entity YAML file with a _Template_ kind and imported into the Backstage catalog.  You can create multiple templates, each of which can perform a different set of steps. For example, you can have one _template_ that creates a React application, and another that creates a serverless app.
+
+Here is an example of a very basic Scaffolder _template_ that prompts the user for a name, and then prints back the text "Hello, name!"
+
+```yaml
+apiVersion: scaffolder.backstage.io/v1beta3
+kind: Template
+metadata:
+  name: hello-world-template
+  title: Hello World
+  description: Says Hello to a specified name.
+spec:
+  owner: backstage/techdocs-core
+  type: service
+
+  parameters:
+    - title: You are about to say hello to your first Backstage Template
+      required:
+        - name
+      properties:
+        name:
+          title: Name
+          type: string
+          description: Name to say hello to
+          ui:autofocus: true
+          ui:options:
+            rows: 5
+
+  steps:
+    - id: log-message
+      name: Log Message
+      action: debug:log
+      input:
+        message: 'Hello, {{ parameters.name }}!'
+```
+
+# Header Section
+
+The header section is required for every _template_ and contains information to configure the task and show details about the task on the "Create Component" page.
+
+```yaml
+apiVersion: scaffolder.backstage.io/v1beta3
+kind: Template
+metadata:
+  name: hello-world-template
+  title: Hello World
+  description: Says Hello to a specified name.
+spec:
+  owner: default/engineering
+  type: service
+```
+
+## apiVersion
+This is a required field and should be set to `scaffolder.backstage.io/v1beta3`
+
+## kind
+A Scaffolder template is also an Entity in Backstage. In order to configure this entity as a template you must set the kind to `Template`
+
+
+## metadata
+The metadata field contains some data that appears on the template card that appears on the "Create Component" page.
+
+## spec
+The spec field contains `owner` and `type`. Owner refers to the Backstage group or user that owns the Scaffolder task e.g. `default/engineering`. Type refers to the type of template. It can be set to anything and appears on the scaffolder template card in the "Create Component" page.
+
+# Parameters
+The parameters is a list of parameter pages that can be prompted from the user when they run a template. Each array element contains the configuration for a single page of items to be filled by the user running the template. The parameter pages must contain `title`, `required` and `properties`.
```

Comment:
```suggestion
The parameters property is a list of parameter pages that can be prompted from the user when they run a template. Each array element contains the configuration for a single page of items to be filled by the user running the template. The parameter pages must contain `title`, `required` and `properties`.
```

Reply:
What is a parameter 'page'?

Reply:
Hey. I think I was mistaken here. I think it's actually an array of parameters which may be an object or other types. Ill check this.